tensorboard/plugins/mesh/mesh_plugin.py

```diff
@@ -194,8 +200,12 @@ def _collect_tensor_events(self, request):
       meta = self._instance_tag_to_metadata[instance_tag]
       tensor_events += [(meta, tensor) for tensor in tensors]
 
-    # Make sure tensors sorted by timestamp in ascending order.
-    tensor_events = sorted(
+    if timestamp is not None:
+      tensor_events = [
+        event for event in tensor_events if event[1].wall_time == timestamp]
+    else:
+      # Make sure tensors sorted by timestamp in ascending order.
+      tensor_events = sorted(
```

Comment:
Likewise, we usually sort by step (the step slider should be monotonic),
unless something special is going on here?